### PR TITLE
Merge dev16.1-preview1 into features/unmanaged-constructed-types

### PR DESCRIPTION
**Pull request by RikkiGibson**
Apologies for the large number of force pushes. Goofed up a few times getting the manual fixups consolidated into one commit.

Please review commit 86f0984.

- Resolved a conflict in MessageID.cs by adding the unmanaged constructed feature ID to the end of the enum declaration.
- Resolved conflicts in SourceNamedTypeSymbol_Bases.cs by using the modified ConstraintsHelper.CheckAllConstraints() that takes a CSharpCompilation
- Resolved conflicts in NullableReferenceTypesTests.cs by changing the expected error spans in a few places as a result of constraint checking being delayed until after type members are added.

### CODE REVIEW

**Comment by RikkiGibson** [comment]
Failure in `Microsoft.CodeAnalysis.CSharp.UnitTests.UnsafeTests.IsUnmanagedTypeSemanticModel`

File: src/Compilers/CSharp/Test/Semantic/Semantics/UnsafeTests.cs [hunk 1]
```diff
@@ -3094,13 +3096,15 @@ class C<T>
             // sbyte, byte, short, ushort, int, uint, long, ulong, char, float, double, decimal, or bool.
             // Any enum_type.
             // Any pointer_type.
-            // Any user-defined struct_type that is not a constructed type and contains fields of unmanaged_types only.
+            // Any user-defined struct_type that contains fields of unmanaged_types only.
```
Comment (RikkiGibson) [inline src/Compilers/CSharp/Test/Semantic/Semantics/UnsafeTests.cs]: @agocke should I send a PR to csharplang to update the spec language over there?

**Comment by jcouv** [comment]
No need for a second review. The code merge was straightforward and tests don't need two reviews.

**Comment by RikkiGibson** [comment]
Thank you!

**Pull request merged**